lib/client-api/src/story_store.js

```diff
@@ -271,6 +280,13 @@ export default class StoryStore extends EventEmitter {
   removeStoryKind(kind) {
     if (this.hasStoryKind(kind)) {
       this._legacydata[toKey(kind)].stories = {};
+
+      this._data = Object.entries(this._data).reduce((acc, [id, story]) => {
```

Comment:
I feel a fear of recursion if both api's call each other..

Reply:
Fortunately they don't -- they just modify the data structures directly?